### PR TITLE
Fix TRUNCATE command for temporary tables.

### PR DESCRIPTION
**Pull request by amosbird**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

### CODE REVIEW

File: dbms/tests/queries/0_stateless/00670_truncate_temporary_table.sql [hunk 1]
```diff
@@ -0,0 +1,16 @@
+drop table if exists test;
+create temporary table test(id int);
+
+select '======Before Truncate======';
+insert into test values(0);
+select * from test;
+
+select '======After Truncate And Empty======';
+truncate table test;
```
Comment (zhang2014) [inline dbms/tests/queries/0_stateless/00670_truncate_temporary_table.sql]: truncate temporary table  test

File: dbms/tests/queries/0_stateless/00670_truncate_temporary_table.sql [hunk 2]
```diff
@@ -0,0 +1,16 @@
+drop table if exists test;
```
Comment (zhang2014) [inline dbms/tests/queries/0_stateless/00670_truncate_temporary_table.sql]: The keyword should be uppercase

Comment (zhang2014) [reply dbms/tests/queries/0_stateless/00670_truncate_temporary_table.sql]: Test in the test database

File: dbms/src/Interpreters/InterpreterDropQuery.cpp [hunk 3]
```diff
@@ -121,18 +121,29 @@ BlockIO InterpreterDropQuery::executeToTemporaryTable(String & table_name, ASTDr
 {
     if (kind == ASTDropQuery::Kind::Detach)
         throw Exception("Unable to detach temporary table.", ErrorCodes::SYNTAX_ERROR);
-    else if (kind == ASTDropQuery::Kind::Drop)
+    else
     {
-        StoragePtr table = (context.hasSessionContext() ? context.getSessionContext() : context).tryRemoveExternalTable(table_name);
+        auto & context_handle = context.hasSessionContext() ? context.getSessionContext() : context;
```
Comment (zhang2014) [inline dbms/src/Interpreters/InterpreterDropQuery.cpp]: Why context_handle ? session_context be better ?

File: dbms/tests/queries/0_stateless/00670_truncate_temporary_table.sql (hunk 2 above)
Comment (amosbird) [reply dbms/tests/queries/0_stateless/00670_truncate_temporary_table.sql]: Well from what I can tell there isn't a consistency or rationale about this.

File: dbms/tests/queries/0_stateless/00670_truncate_temporary_table.sql (hunk 1 above)
Comment (amosbird) [reply dbms/tests/queries/0_stateless/00670_truncate_temporary_table.sql]: `temporary` isn't needed if no database prefix is specified.

Comment (zhang2014) [reply dbms/tests/queries/0_stateless/00670_truncate_temporary_table.sql]: You can refer to https://github.com/zhang2014/ClickHouse/blob/960fad130ca68dc6e9e5fb282a541d3a780c2200/dbms/tests/queries/0_stateless/00564_show_tables_with_temporary.sql, They should be consistent : )

File: dbms/tests/queries/0_stateless/00670_truncate_temporary_table.sql (hunk 2 above)
Comment (zhang2014) [reply dbms/tests/queries/0_stateless/00670_truncate_temporary_table.sql]: Sure, but I think it would be better : )

**Pull request merged**